api/python/provisioner/commands/__init__.py

```diff
@@ -351,6 +352,138 @@ def run(self, *args, targets=ALL_MINIONS, **kwargs):
+                StatesApplier.apply(self.post_states)
+            except Exception:
+                logger.warning(
+                    "Starting Rollback: Result of "
+                    "failure to apply pillar changes."
+                )
+                # TODO more solid rollback
+                pillar_updater.rollback()
+                pillar_updater.apply()
+                raise
+        except Exception:
+            logger.exception('Failed to apply changes')
+            # treat post as restoration for pre, apply
+            # if rollback happened
+            StatesApplier.apply(self.post_states)
+            raise
+
+    @staticmethod
+    def set_ctrl_ntp():
+        script = (
+            PRVSNR_FILEROOT_DIR /
+            'components/controller/files/scripts/controller-cli.sh'
+        )
+
+        # TODO: Update logic to find enclosure_N based on current node_id
+        enclosure = "enclosure-1"
+        enclosure_pillar_path = KeyPath(f"storage/{enclosure}")
+        system_pillar_path = KeyPath("system/ntp")
+        ip = PillarKey(enclosure_pillar_path / 'controller/primary/ip')
+        user = PillarKey(enclosure_pillar_path / 'controller/user')
+        passwd = PillarKey(enclosure_pillar_path / 'controller/secret')
+
+        ntp_server = PillarKey(system_pillar_path / 'time_server')
+        timezone = PillarKey(system_pillar_path / 'time_zone')
+
+        pillar = PillarResolver(LOCAL_MINION).get([ip, user, passwd, ntp_server, timezone])
+        pillar = next(iter(pillar.values()))
+
+        for param in (ip, user, passwd, ntp_server, timezone):
+            if not pillar[param] or pillar[param] is values.MISSED:
+                raise BadPillarDataError(
+                    'value for {} is not specified'.format(param.keypath)
+                )
+
+        logger.info("Initiating NTP configuration on storage enclosure")
+        logger.debug("Calling controller-cli utility to set NTP")
+        try:
+            StateFunExecuter.execute(
+                'cmd.run',
+                fun_kwargs=dict(
+                    name=(
+                        "{script} host -h {ip} -u {user} -p {passwd} "
+                        "--ntp {ntp_server} '{timezone}'"
+                        .format(
+                            script=script,
+                            ip=pillar[ip],
+                            user=pillar[user],
+                            passwd=pillar[passwd],
+                            ntp_server=pillar[ntp_server],
+                            timezone=pillar[timezone]
+                        )
+                    )
+                )
+            )
+        except Exception:
+            logger.exception('Failed to configure NTP on enclosure')
+            raise
+
+    def _run(self, params, targets):
+        self._apply(params, targets)
+        # Configure NTP only if the setup type is physical
+        from .grains_get import GrainsGet
+        from ..salt import local_minion_id
```

Comment:
@83bhp  Can we add imports at the beginning and local_minion_id is already imported in this file.
https://github.com/Seagate/cortx-prvsnr/blob/pre-cortx-1.0/api/python/provisioner/commands/__init__.py#L74

Reply:
thanks for pointing, addressed!